lib/ansible/modules/cloud/amazon/s3.py

```diff
@@ -713,22 +713,18 @@ def main():
 def get_s3_connection(aws_connect_kwargs, location, rgw, s3_url):
     if s3_url and rgw:
         rgw = urlparse(s3_url)
-        s3 = boto.connect_s3(
-            is_secure=rgw.scheme == 'https',
-            host=rgw.hostname,
-            port=rgw.port,
-            calling_format=OrdinaryCallingFormat(),
-            **aws_connect_kwargs
-        )
+        aws_connect_kwargs['is_secure'] = rgw.scheme == 'https'
```

Comment:
I think it might be better to only minimally change the code by changing line [529](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/s3.py#L529) to `if '.' in bucket and not rgw and not is_fakes3(s3_url):` instead of manually fixing aws_connect_kwargs.